source/slang/slang-emit-glsl.cpp

```diff
@@ -310,7 +334,16 @@ void GLSLSourceEmitter::_emitGLSLParameterGroup(IRGlobalParam* varDecl, IRUnifor
     or IRGLSLShaderStorageBufferType which is read write.
     */
 
-    _emitGLSLLayoutQualifier(LayoutResourceKind::DescriptorTableSlot, &containerChain);
+    {
+        // TODO(JS):
+        // The *assumption* here is that we only need to detect the LayoutResourceKind as used on the initial containerChain.varLayout
+        // rather than search up the chain. 
+        //
+        // Perhaps there is still an issue here, because perhaps it's possible (?) to have a mixture of ConstantBuffer/DescriptorSlot
+        // and in that case the usedKind would just restrict to one and so produce the wrong answer.
+        const auto usedKind = _findUsedDescriptorSlotLikeKind(containerChain.varLayout, LayoutResourceKind::ConstantBuffer);
+        _emitGLSLLayoutQualifier(usedKind, &containerChain);
+    }
     _emitGLSLLayoutQualifier(LayoutResourceKind::PushConstantBuffer, &containerChain);
```

Comment:
IIRC, the two different calls to `_emitGLSLLayoutQualifier` in the original code were basically there to address the same kind of concern as this PR: a `ConstantBuffer<...>` might be allocated to use either a `DescriptorTableSlot` *or* a `PushConstantBuffer` for Vulkan.

It might be possible to further clean up this code by making the core `_findUsedDescriptorSlotLikeKind` operation take an array of `LayoutResourceKind`s to iterate over and check for (in order), so that this call site could pass in the three kinds it cares about: `ConstantBuffer`, `PushConstantBuffer`, and `DescriptorTableSlot`.

(A wrapper around that core version of `_findUsedDescriptorSlotLikeKind` could then cover the other call sites from this PR...)